File: Pertemuan 11/membuatarraynim.php
<?php
25
$nim = readline("Masukkan NIM: ");

$digit_string = str_split($nim); 
$digit = [];
foreach ($digit_string as $d) {
    $digit[] = (int)$d; 
}

echo "\nArray Digit: " . implode(", ", $digit) . "\n";

$total_digit = 0;
$maksimum_digit = -1; 
$minimum_digit = 10;
$jumlah_elemen = count($digit);

foreach ($digit as $d) {
    $total_digit += $d;
    
    if ($d > $maksimum_digit) {
        $maksimum_digit = $d;
    }
        
    if ($d < $minimum_digit) {
        $minimum_digit = $d;
    }
}

if ($jumlah_elemen > 0) {
    $rata_rata_digit = $total_digit / $jumlah_elemen;
} else {
    $rata_rata_digit = 0;
}

$reverse_array = [];
for ($i = $jumlah_elemen - 1; $i >= 0; $i--) {
    $reverse_array[] = $digit[$i];
}


echo "\n--- Hasil Perhitungan ---\n";
echo "Total seluruh digit: " . $total_digit . "\n";
echo "Nilai maksimum digit: " . $maksimum_digit . "\n";
echo "Nilai minimum digit: " . $minimum_digit . "\n";
echo "Nilai rata-rata digit: " . number_format($rata_rata_digit, 2) . "\n"; 
echo "Reverse array: " . implode(", ", $reverse_array) . "\n";

?>

File: Pertemuan 11/membuatarraynim.java
import java.util.Scanner;
import java.util.Arrays;

public class membuatarraynim {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

       
        System.out.print("Masukkan NIM:");
        String nimStr = scanner.nextLine();
        scanner.close();

        
        int[] digit = new int[nimStr.length()];
        for (int i = 0; i < nimStr.length(); i++) {
           
            digit[i] = Character.getNumericValue(nimStr.charAt(i)); 
        }

        System.out.println("Digit Array: " + Arrays.toString(digit));

    
        int total = 0;
        for (int d : digit) {
            total += d;
        }

        
        int maks = digit[0];
        int minim = digit[0];

        for (int d : digit) {
            if (d > maks) {
                maks = d;
            }
            if (d < minim) {
                minim = d;
            }
        }

       
        double rata = (double) total / digit.length;

       
        int[] rev = new int[digit.length];
        for (int i = 0; i < digit.length; i++) {
            rev[i] = digit[digit.length - 1 - i];
        }

        
        System.out.println("Total: " + total);
        System.out.println("Maksimum: " + maks);
        System.out.println("Minimum: " + minim);
        System.out.println("Reverse Array: " + Arrays.toString(rev));
    }
}